synapse/handlers/room_member.py

```diff
@@ -424,7 +422,7 @@ async def _local_membership_update(
         # do it up front for efficiency.)
         if txn_id:
             existing_event_id = None
-            if self._msc3970_enabled and requester.device_id:
+            if requester.device_id:
```

Comment:
Is it still possible to have requesters without device IDs? I think so...

Reply:
Although this seems to conflict with below: https://github.com/matrix-org/synapse/pull/15629/files#r1198058261